oidc-consumer/test/index.test.ts

```diff
@@ -0,0 +1,112 @@
+import OidcConsumer from "../dist/cjs/index";
+import * as sinon from 'sinon';
+import { mockReq, mockRes } from 'sinon-express-mock';
+const realm = process.env.REALM || 'default_realm'; 
+
+const allowedOrigins: string[] = (process.env.ALLOWED_ORIGINS?.split(",") || []);
+const allowedRedirectURIs: string[] = (process.env.ALLOWED_REDIRECT_URIS?.split(",") || []);
+
+const consumer = new OidcConsumer({
+    scope: "openid profile email",
+    callback_route: `/${realm}/callback`,
+    allowedRedirectURIs,
+    clientConfig: {
+        client: {
+            id: process.env.CLIENT_ID || 'your-client-id',
+            secret: process.env.CLIENT_SECRET || 'your-client-secret',
+        },
+        auth: {
+            tokenHost: 'https://your-token-host-url.com',
+            tokenPath: `/auth/realms/${realm}/protocol/openid-connect/token`,
+            revokePath: `/auth/realms/${realm}/protocol/openid-connect/logout`,
+            authorizePath: `/auth/realms/${realm}/protocol/openid-connect/auth`,
+        },
+        options: {
+            authorizationMethod: "body",
+        },
+    },
+    sessionOptions: {
+        store: new FirestoreStore({
+            dataset: new Firestore(),
+            kind: 'test',
+        }),
+        secret: process.env.SESSION_SECRET || 'your-session-secret', 
+        resave: false,
+        saveUninitialized: false,
+    },
+});
+
+describe('Authentication Functions', () => {
+  let firestoreStub=sinon.SinonStub;
+
+  beforeEach(() => {
+    firestoreStub = sinon.stub(Firestore.prototype, 'collection').returnsThis();
+    firestoreStub.doc = sinon.stub().returnsThis();
+    firestoreStub.get = sinon.stub();
+  });
+
+  afterEach(() => {
+    firestoreStub.restore();
+  });
+
+  describe('verifySession', () => {
+    it('should call next() if session state is present', async () => {
+      const req = mockReq();
+      const res = mockRes();
+      req.session = { state: 'dummy state../dist/cjs/src/index', reload: jest.fn()};
+
+      const next = sinon.spy();
+
+      await consumer.verifySession(req, res, next);
+
+      expect(next.calledOnce).toBe(true);
+    });
+
+    it('should throw an error if session state is not present and throwError is true', async () => {
+      const req = mockReq();
+      const res = mockRes();
+      req.session = {reload: jest.fn()};
+
+      const next = sinon.spy();
+
+        await consumer.verifySession(req, res, next, true);
+        expect(next.calledOnce).toBe(true);
+        expect(next.firstCall.args[0]).toBeInstanceOf(Error);
+        expect(next.firstCall.args[0].message).toEqual('SESSION_VERIFICATION_FAILED');
+    });
+  });
+
+  describe('authCallback', () => {
+    it('should call next() if state matches session state', async () => {
+      const req = mockReq({
+        query: { code: 'dummy_code', state: 'dummy_state' },
+      });
+      const res = mockRes();
+      req.session = { state: 'dummy_state' };
```

Comment:
not sure how this works as state is never sent as query param (exposed to client)
need to modify firestore document

Reply:
Removed the query param, wasn't necessary. 
Since firestore is specific to our use case it doesn't make sense adding a firestore specific test case in the library. So the req.session.reload is just mocked. This tests the logic in terms of the flow we want to implement.